docs/src/main/sphinx/security/oauth2.rst

```diff
@@ -0,0 +1,126 @@
+========================
+OAuth 2.0 authentication
+========================
+
+Trino can be configured to enable OAuth 2.0 authentication over HTTPS for the
+Web UI and the JDBC driver. Trino use `Authorization Code
+<https://tools.ietf.org/html/rfc6749#section-1.3.1>`_ flow which exchanges an
+Authorization Code for a token. The flow includes, at a high level, following
+steps:
+
+#. the Trino coordinator redirects a user agent to the Authorization Server
+#. the user authenticates in the Authorization Server and approves the Trino's permissions request
```

Comment:
authenticates with the Authorization Server, and it approves Trino's permission request